events/deployment_info.go

```diff
@@ -0,0 +1,122 @@
+package events
+
+import (
+	"encoding/json"
+	"github.com/allegro/marathon-consul/apps"
+)
+
+type DeploymentEvent struct {
+	Type        string       `json:"eventType"`
+	Plan        *Plan        `json:"plan"`
+	CurrentStep *CurrentStep `json:"currentStep"`
+}
+
+type Plan struct {
+	Original *Deployments `json:"original"`
+	Target   *Deployments `json:"target"`
+}
+
+type Deployments struct {
+	Apps []*apps.App `json:"apps"`
+}
+
+type CurrentStep struct {
+	Actions []*Action `json:"actions"`
+}
+
+type Action struct {
+	Type  string     `json:"type"`
+	AppId apps.AppId `json:"app"`
+}
+
+func (d *DeploymentEvent) OriginalApps() []*apps.App {
+	if d.Plan != nil {
+		if d.Plan.Original != nil {
+			if d.Plan.Original.Apps != nil {
+				return d.Plan.Original.Apps
+			}
+		}
+	}
+	return []*apps.App{}
+}
+
+func (d *DeploymentEvent) TargetApps() []*apps.App {
+	if d.Plan != nil {
+		if d.Plan.Target != nil {
+			if d.Plan.Target.Apps != nil {
+				return d.Plan.Target.Apps
+			}
+		}
+	}
+	return []*apps.App{}
+}
+
+func (d *DeploymentEvent) Actions() []*Action {
+	if d.CurrentStep != nil {
+		if d.CurrentStep.Actions != nil {
+			return d.CurrentStep.Actions
+		}
+	}
+	return []*Action{}
+}
+
+func (d *DeploymentEvent) StoppedConsulApps() []*apps.App {
+	return d.consulAppsWithActionPerformed(d.OriginalApps(), "StopApplication")
```

Comment:
Maybe we shoud extract actions to some const and link them with Marathon code https://github.com/mesosphere/marathon/blob/e3e867f00d4964d0367ec7e07519d6c9545681d6/src/main/scala/mesosphere/marathon/upgrade/DeploymentPlan.scala#L19-L34